### ISSUE TITLE
UIChatHandler: MissingReferenceException after using Warp Portals

### ISSUE THREAD

**Issue by shubhank008** [suriyun-production/mmorpg-kit-docs#1131]
Using Unity 2021.2
Using MMORPGKit 1.69 (latest one)  

When using warp portals to warp, you get a MissingReferenceException for the UIChatHandler  
After that every time someone sends a chat message, that exception is thrown as if the game lost a reference to the chat, however, other clients (who have not warped) can still receive messages if I send from my side.  

For the time being, I added a ```if(gameObject != null)``` check to suppress the exceptions  

<img width="1098" alt="Unity_UOK38GUnq6" src="https://user-images.githubusercontent.com/3790163/132981343-4f3e1a37-0326-4a4b-9ece-5d2226802efb.png">

**Comment by insthync** [comment]
Okay, I will add null checking too.

**Comment by BJPickles** [comment]
Just to mention that this is happening on scene change.

So logging in, then logging back out (but keeping the client still open) will throw the error.

**Comment by insthync** [comment]
Add this to `OnReceiveChat`

```
            if (this == null)
            {
                RemoveOnClientReceiveChatMessage();
                return;
            }
```

![Screenshot_646](https://user-images.githubusercontent.com/2697550/133869707-ff69d0b7-2d4a-475d-950c-6fe7b136e026.png)

**Comment by BJPickles** [comment]
> Add this to `OnReceiveChat`
> 
> ```
>             if (this == null)
>             {
>                 RemoveOnClientReceiveChatMessage();
>                 return;
>             }
> ```
> 
> ![Screenshot_646](https://user-images.githubusercontent.com/2697550/133869707-ff69d0b7-2d4a-475d-950c-6fe7b136e026.png)

This worked, thank you!

**Comment by insthync** [comment]
Fixed in 1.69c
